### PR TITLE
Bitset iteration micro optimize

### PR DESCRIPTION
**Pull request by corona10**
- Bitset iteration micro optimize

### CODE REVIEW

File: src/core/util.h [hunk 1]
```diff
@@ -153,19 +153,18 @@ template <int N> struct BitSet {
...
             assert(cur >= 0 && cur < N);
             uint16_t tmp = set.bits;
             tmp >>= cur + 1;
             cur++;
-
-            while (cur < N) {
-                if (tmp & 1)
+            if (tmp > 0) {
+                int offset = __builtin_ctz(tmp);
+                if (offset < N) {
```
Comment (kmod) [inline src/core/util.h]: Should be `cur + offset < N`?

File: src/core/util.h [hunk 2]
```diff
@@ -153,19 +153,18 @@ template <int N> struct BitSet {
...
+                    cur += offset;
                     return *this;
-
```
Comment (corona10) [inline src/core/util.h]: @kmod Yes you're right. it's my fault. I will fix it

**Comment by corona10** [comment]
@kmod I fixed it :-)

**Comment by kmod** [comment]
awesome thanks!

**Pull request merged**